### PR TITLE
chore(ST-56): integrate the combined reg api url in frontend

### PR DESCRIPTION
**Pull request by faysal-alam-sazim**
## Description of Change

- Removed two different reg mutations
- Added a mutation called register 
- Removed two different dtos for creating student and teacher
- Updated IUserDto
- Called the new api from the student and teacher registration hook    

## Associated Ticket Link(s)

- https://trello.com/c/WsPtywCY

## Related Pull Request(s)

- N/A

### CODE REVIEW

File: modules/Register/hooks/useTeacherRegistration.ts [hunk 1]
```diff
@@ -40,7 +40,7 @@ const useTeacherRegistration = () => {
     };
 
     try {
-      const res = await registerTeacher(newTeacher).unwrap();
+      const res = await register(newTeacher).unwrap();
```
Comment (ishmam-iqbal-sazim) [inline modules/Register/hooks/useTeacherRegistration.ts:43]: Let's add new line below

File: modules/Register/hooks/useStudentRegistration.ts [hunk 2]
```diff
@@ -43,7 +43,7 @@ const useStudentRegistration = () => {
     };
 
     try {
-      const res = await registerStudent(newStudent).unwrap();
+      const res = await register(newStudent).unwrap();
```
Comment (ishmam-iqbal-sazim) [inline modules/Register/hooks/useStudentRegistration.ts:46]: Let's add new line below

File: shared/redux/rtk-apis/users/users.api.ts [hunk 3]
```diff
@@ -16,31 +16,16 @@ const usersApi = projectApi.injectEndpoints({
       transformResponse: (response: TApiResponse<TUser[]>) => response.data,
     }),
 
-    registerStudent: builder.mutation<TLoginResponse, ICreateStudentDto>({
-      query: (newStudent) => ({
-        url: "users/signup/student",
+    register: builder.mutation<TLoginResponse, IUserDto>({
+      query: (newUser) => ({
+        url: "users/signup",
```
Comment (ishmam-iqbal-sazim) [inline shared/redux/rtk-apis/users/users.api.ts:21]:
```suggestion
        url: "auth/signup",
```

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Approved pending comments

**Review by ragib-hossain-sazim — APPROVED** [review]
Approved pending passing lint

File: shared/redux/rtk-apis/users/users.api.ts (hunk 3 above)
Comment (faysal-alam-sazim) [reply shared/redux/rtk-apis/users/users.api.ts:21]: have a ticket to move all the registration endpoinds from user to auth module

**Pull request merged**